Common/src/containers/CLookUpTable.cpp

```diff
@@ -79,32 +120,67 @@ void CLookUpTable::LoadTableRaw(const string& var_file_name_lut) {
     cout << "Loading lookup table, filename = " << var_file_name_lut << " ..." << endl;
 
   file_reader.ReadRawLUT(var_file_name_lut);
-
-  n_points = file_reader.GetNPoints();
-  n_triangles = file_reader.GetNTriangles();
+  table_dim = file_reader.GetTableDim();
+  n_table_levels = file_reader.GetNLevels();
+
+  n_points.resize(n_table_levels); 
+  n_triangles.resize(n_table_levels);
+  n_hull_points.resize(n_table_levels);
+  table_data.resize(n_table_levels);
+  hull.resize(n_table_levels);
+  triangles.resize(n_table_levels);
+  interp_mat_inv_x_y.resize(n_table_levels);
+  edges.resize(n_table_levels);
+  edge_to_triangle.resize(n_table_levels);
+
+  for(unsigned long i_level=0; i_level<n_table_levels; i_level++){
+    n_points[i_level] = file_reader.GetNPoints(i_level);
+    n_triangles[i_level] = file_reader.GetNTriangles(i_level);
+    n_hull_points[i_level] = file_reader.GetNHullPoints(i_level);
+    table_data[i_level] = file_reader.GetTableData(i_level);
+    triangles[i_level] = file_reader.GetTriangles(i_level);
+    hull[i_level] = file_reader.GetHull(i_level);
+  }
+  
   n_variables = file_reader.GetNVariables();
-  n_hull_points = file_reader.GetNHullPoints();
   version_lut = file_reader.GetVersionLUT();
   version_reader = file_reader.GetVersionReader();
   names_var = file_reader.GetNamesVar();
-  table_data = file_reader.GetTableData();
-  triangles = file_reader.GetTriangles();
-  hull = file_reader.GetHull();
 
+  if(table_dim == 3){
+    z_values_levels.resize(n_table_levels);
+    for(unsigned long i_level=0; i_level<n_table_levels; i_level++){
+      z_values_levels[i_level] = file_reader.GetTableLevel(i_level);
+    }
+  }
   if (rank == MASTER_NODE) 
     cout << " done." << endl;
 }
 
-void CLookUpTable::FindTableLimits(const string& name_prog, const string& name_enth) {
-  int ixEnth = GetIndexOfVar(name_enth);
-  int ixProg = GetIndexOfVar(name_prog);
-
-  /* we find the lowest and highest value of enthalpy and progress variable in the table */
-  limits_table_enth[0] = *min_element(&table_data[ixEnth][0], &table_data[ixEnth][0]+table_data.cols());
-  limits_table_enth[1] = *max_element(&table_data[ixEnth][0], &table_data[ixEnth][0]+table_data.cols());
-  limits_table_prog[0] = *min_element(&table_data[ixProg][0], &table_data[ixProg][0]+table_data.cols());
-  limits_table_prog[1] = *max_element(&table_data[ixProg][0], &table_data[ixProg][0]+table_data.cols());
+void CLookUpTable::FindTableLimits(const string& name_cv1, const string& name_cv2) {
+  int idx_X = GetIndexOfVar(name_cv1);
+  int idx_Y = GetIndexOfVar(name_cv2);
```

Comment:
Where possible, I think we should use the naming convention "cv1,cv2,cv3" for the controlling variables and avoid additional names for the manifold dimensions like X,Y,Z. At the level of the LUT we can indeed avoid the specific words  "enthalpy, progress variable, mixture fraction"
```suggestion
  int idx_cv1 = GetIndexOfVar(name_cv1);
  int idx_cv2 = GetIndexOfVar(name_cv2);
```